File: weeak-1/maximumPairWiseProduct-o(n)/maximumPairWiseProduct.js
const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function sumProduct(arr){
	let max1 = -1;
	let max2 = -1;

	for(let i=0; i<arr.length;i++){
		if(arr[i] > max1){
			max1 = arr[i]
		}
	}

	for(let j=0; j<arr.length;j++){
		if(arr[j] > max2 && arr[j] != max1){
			max2 = arr[j]
		}
	}
	return max1 * max2
}

rl.question("Enter array of integers ", function(arr) {
	arr = arr.split(" ")
	arr = arr.map(x => parseInt(x))
	console.log(sumProduct(arr))

});


rl.on("close", function() {
    console.log("\nBYE BYE !!!");
    process.exit(0);
});

File: weeak-1/maximumPairWiseProduct-o(n)/maximumPairWiseProduct.cpp
#include <iostream>
#include <vector>

using namespace std;
long long sumProduct(vector<long>);

int main(){
	int n;
	vector<long> b(n);

	cout << "Enter your array length " << endl;
	cin>>n;

	for(int i=0; i<n; i++){
		cin>>b[i];
	}
	cout << sumProduct(b) << endl;
	return 0;
}

long long sumProduct(vector<long> b){
	int sizee = b.size();
	long max1 = -1;
	long max2 = -1;
	for (int i=0; i<sizee;i++){
		if(b[i] > max1){
			max1 = b[i];
		}
	}

	for (int j=0; j<sizee;j++){
		if(b[j] > max2 && b[j] != max1){
			max2 = b[j];
		}
	}
	
	return ((long long)max1 * max2);
}
